### DISCUSSION TITLE
Can use `~` to replace home path

### DISCUSSION THREAD

**Discussion by Buer-Nahida** [YaLTeR/niri#226] [Ideas]
Use
```
SUPER+Alt+J { spawn "~/.config/niri/scripts/control/volume" "--dec"; }
```
to replace
```
SUPER+Alt+J { spawn "/home/xxx/.config/niri/scripts/control/volume" "--dec"; }
```
I know I can use it, but my volume script dosen't work if I use it:
```
SUPER+Alt+J { spawn "sh" "-c" "~/.config/niri/scripts/control/volume" "--dec"; }
```

**Comment by YaLTeR** [comment]
What if you do `spawn "sh" "-c" "~/.config/niri/scripts/control/volume --dec"`? Last argument as a whole.

I suppose we could add a similar simple replacement to how screenshot-path works, that would be consistent.

**Reply by Buer-Nahida** [reply]
Yes, it is work on my linux, thx.

**Comment by Buer-Nahida** [comment]
Finished

**Comment by Buer-Nahida** [comment]
Reopen, because niri can't use
```kdl
spawn-at-startup "swaybg" "-i" "~/.config/niri/wallpaper/wallpaper.png"
```
to replace
```kdl
spawn-at-startup "swaybg" "-i" "/home/xxx/.config/niri/wallpaper/wallpaper.png"
```

**Reply by YaLTeR** [reply]
I'd like to avoid replacing it in arguments. The command is a path, but the arguments can be anything, and always parsing them as paths isn't really right.

**Reply by Buer-Nahida** [reply]
Yes，but maybe we can execute these commands use shell？Such as default use `sh -c '[command]'` to execute commands，I tested my all shell scripts and commands，use `sh -c` prefix to execute it are haven't different with don't use.

If some users want change default shell，they can use a option to change shell，such as
```kdl
executer "sh" "-c"
```

**Reply by YaLTeR** [reply]
I'll need to check what other compositors do I guess. One thing is that `sh` has a little bit of overhead compared to running the command directly.

**Reply by Buer-Nahida** [reply]
I have a point, If a `\` in front of `~` then do nothing, else parsing `~`.
